soci/soci_index.go

```diff
@@ -250,53 +249,62 @@ func ParseOptimization(s string) (Optimization, error) {
 	return "", fmt.Errorf("optimization %s is not a valid optimization %v", s, Optimizations)
 }
 
-// BuildOption specifies a config change to build soci indices.
-type BuildOption func(c *buildConfig) error
+// BuilderOption is a functional argument that affects a SOCI index builder
+// and all indexes built with that builder.
+type BuilderOption func(c *builderConfig) error
```

Comment:
I know it is pre-existing but do any of the build options return an error?

Reply:
No. The thought was that it's easier to ignore it than to add it later, but I guess since `builderConfig` is not exported it doesn't really matter.